app/src/main/java/com/epmedu/animeal/component/IntentGpsSettingsProvider.kt

```diff
@@ -0,0 +1,48 @@
+package com.epmedu.animeal.component
+
+import android.content.Context
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import android.location.GnssStatus
+import android.location.LocationManager
+import android.os.Handler
+import android.os.Looper
+import android.provider.Settings
+import com.epmedu.animeal.common.component.GpsSettingsProvider
+import kotlinx.coroutines.channels.ProducerScope
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
+
+/** Allows to enable/disable gps settings via intent call to the system */
+class IntentGpsSettingsProvider(private val androidApplicationContext: Context) : GpsSettingsProvider {
+
+    override val isGpsSettingsEnabled: Boolean
+        get() = androidApplicationContext.getLocationManager()?.isProviderEnabled(LocationManager.GPS_PROVIDER) ?: false
+
+    // FLAG_ACTIVITY_NEW_TASK is required when we're starting activity from app context
+    override fun changeGpsSettings() = androidApplicationContext.startActivity(
+        Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS).setFlags(FLAG_ACTIVITY_NEW_TASK)
+    )
+
+    override fun fetchGpsSettingUpdates() = callbackFlow { suspendFetchGpsSettingUpdates() }
+
+    private suspend fun ProducerScope<Boolean>.suspendFetchGpsSettingUpdates() {
+        val locationManager = androidApplicationContext.getLocationManager() ?: return
+
+        val callback = object : GnssStatus.Callback() {
```

Comment:
<img width="944" alt="image" src="https://user-images.githubusercontent.com/16294951/191242899-29cdae62-758a-4adc-90ca-785551d3bdd3.png">

added in API 24 🙁 

we can add something like this
https://github.com/JacekPietras/ZOO/blob/master/tracking/src/main/java/com/jacekpietras/zoo/tracking/listener/GpsStatusListenerCompat.kt

Reply:
Well, at the end of the provided file GnssStatus already used...

Reply:
But it checks SDK version

```Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ```